datafusion/core/src/physical_optimizer/enforcement.rs

```diff
@@ -885,14 +915,9 @@ fn ensure_distribution_and_ordering(
                 Ok(child)
             } else {
                 let sort_expr = required.unwrap().to_vec();
-                if child.output_partitioning().partition_count() > 1 {
-                    Ok(Arc::new(SortExec::new_with_partitioning(
-                        sort_expr, child, true, None,
-                    )) as Arc<dyn ExecutionPlan>)
-                } else {
-                    Ok(Arc::new(SortExec::try_new(sort_expr, child, None)?)
-                        as Arc<dyn ExecutionPlan>)
-                }
+                Ok(Arc::new(SortExec::new_with_partitioning(
```

Comment:
My understand is that the `SortExecs` added by the enforcement rule should be always local Sort. Because in the `ExecutionPlan` Trait, `output_ordering()` method specifies how data is ordered in each partition, and the `required_input_orderings()`  method specifies the sort order requirements for each partition. 

@yahoNanJing 
I think maybe you could add some comments to the `ExecutionPlan` methods and here to make it clear.

Reply:
yes. I'll add more comments

Reply:
Yes. I'll add some comments there. Otherwise, it may be misleading.